### PR TITLE
Provide an individual color temperature range per model

### PR DESCRIPTION
**Pull request by syssi**
cp. https://github.com/home-assistant/home-assistant/pull/14083/

### CODE REVIEW

File: yeelight/main.py [hunk 1]
```diff
@@ -21,6 +21,19 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+MODEL_SPECS = {
```
Comment (skorokithakis) [inline yeelight/main.py]: Can you add a leading underscore here?

File: yeelight/main.py [hunk 2]
```diff
@@ -183,6 +196,7 @@ def __init__(self, ip, port=55443, effect="smooth",
         self.duration = duration
         self.auto_on = auto_on
         self.power_mode = power_mode
+        self.model = model
```
Comment (skorokithakis) [inline yeelight/main.py:204]: Hmm, where is this documented?

**Comment by skorokithakis** [comment]
Thanks for your PR! I've left some comments.

**Comment by skorokithakis** [comment]
Alright, this looks reasonable. Just to clarify, we don't get the model from the bulb, right? The user has to specify their model in order to get the model-specific parameters, is that correct?

**Comment by syssi** [comment]
The model is part of the `discover_bulbs` response. This (SSDP) is the only source to retrieve the information from the bulb.

**Comment by skorokithakis** [comment]
Hmm I see, thank you. The PR looks good overall, I'm just wondering whether `get_color_temp_range` is appropriate, because I can see this being used to get other ranges in the future. Maybe we should call it `get_model_ranges` and return a dictionary like `{"color_temp": {"min": 1200, "max": 6500}}`?

Also, since this will need to be retrieved from `discover_bulbs` and set explicitly, I wonder if we should have a method like `discover_bulbs` that will return `Bulb` instances (with the `model` already set) instead of just dictionaries, but that's for another PR.

File: yeelight/main.py [hunk 3]
```diff
@@ -362,6 +379,20 @@ def set_color_temp(self, degrees, **kwargs):
...
 
+    @_command
+    def get_color_temp_range(self, **kwargs):
+        """
+        Return the bulb's color temperature range.
+        """
+        if self.model is not None and self.model in _MODEL_SPECS:
+            return _MODEL_SPECS[self.model]
+
+        if self.bulb_type is BulbType.White:
```
Comment (skorokithakis) [inline yeelight/main.py]: By the way, there's now a bulb type called `WhiteTemp`, for white bulbs that support temperature adjustments.

**Comment by syssi** [comment]
I will care about the changes ASAP.

**Comment by quthla** [comment]
Is this okay to merge now with the recent changes?

**Comment by skorokithakis** [comment]
Yes, sorry for the delay. It should be good, thanks!

**Comment by syssi** [comment]
Thanks! Could you prepare a release? I would like to introduce the new feature in HA.

**Comment by skorokithakis** [comment]
I will, I'm just thinking about whether to make the API a bit more general. Can you bug me again in a few days if it's not ready by then?

**Comment by syssi** [comment]
Sure!

**Comment by quthla** [comment]
Is there any news on the release yet?

**Comment by skorokithakis** [comment]
Sorry, I've been meaning to clean up the API and release, but I've been too busy this past few days. I'll do it ASAP.

**Comment by skorokithakis** [comment]
`0.4.1` released, thank you all!

**Pull request merged**